### PR TITLE
User list

### PR DESCRIPTION
**Pull request by Max-Hanyn**

### CODE REVIEW

**Comment by Max-Hanyn** [comment]
pull reauest

File: .htaccess [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#</Files>
+
+#<FilesMatch "\autoload.php$">
+#    Order Allow,Deny
+#    Deny from all
+#</FilesMatch>
+#<FilesMatch "index[0-9]?\autoload.php$">
+#    Order Allow,Deny
+#    Allow from all
+#</FilesMatch>
```
Comment (dankolosov) [inline .htaccess]: Remove commented code

File: Classes/App.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+class App
```
Comment (dankolosov) [inline Classes/App.php]: remove unused class

File: Classes/Mail.php [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+
+class Mail
```
Comment (dankolosov) [inline Classes/Mail.php]: remove unused class

File: Classes/Roles.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        RolesModel::ROLE_USER_ID => [RolesModel::ROLE_USER_ID,RolesModel::ROLE_GUEST_ID],
+        RolesModel::ROLE_MODERATOR_ID => [RolesModel::ROLE_MODERATOR_ID,RolesModel::ROLE_USER_ID,RolesModel::ROLE_GUEST_ID],
+        ];
+    return $roles[$roleId];
+}
+
+    static public function checkRole($role)
+    {
+//        print_r($_SESSION);
+////        exit();
```
Comment (dankolosov) [inline Classes/Roles.php]: remove commented code

File: Classes/Roles.php [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+
+        $roles = (new Roles)->getRole( $_SESSION['user']['roleId']);
```
Comment (dankolosov) [inline Classes/Roles.php]: $this->getRole()

File: Classes/Roles.php [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+
+        foreach ($roles as $permission) {
```
Comment (dankolosov) [inline Classes/Roles.php]: replace to 
return in_array($role, $this->getRole($_SESSION['user']['roleId']))

File: Classes/Route.php [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+//
+//        echo $i. " ". $j;
+//    }
+//     public static function group($params,$func){
+//
+//        call_user_func($func());
+//
+//
+//
+//     }
```
Comment (dankolosov) [inline Classes/Route.php]: remove commented code

File: Classes/Route.php [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+                    $routeAndParams[$i] = str_replace("$key", $param, $routeAndParams[$i], $count);
+                    if ($count > 0) {
+                        array_push($parameters, $i);
+                    }
+                }
+            }
+            $route = implode('/',$routeAndParams);
+            $test = [];
+            array_push($test,$route,$parameters);
+            return $test;
```
Comment (dankolosov) [inline Classes/Route.php]: change the name of the test variable.

File: Classes/Route.php [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+            $key = key($route);
+
+
+
+            if (preg_match("#\A$key\z#", $_GET['url'])) {
+//                    print_r($_SESSION);
+//                        print_r($route[$key]['roleId']);
+//                    exit();
+
+               // echo "f";
```
Comment (dankolosov) [inline Classes/Route.php]: remove commented code

File: Controllers/AdminController.php [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+    }
+
+
+    $user = $userProfileModel->getProfile($id);
+    $this->view('adminEdit',['user' => $user,'massage' => $massage]);
+}
+public function changeRole(){
+    $userModel = new UserModel();
+    $userModel->update(['role_id' => $_POST['role']])->where('id',$_POST['userId'])->execute();
```
Comment (dankolosov) [inline Controllers/AdminController.php]: check if $_POST['role'] exists

File: Controllers/HomeController.php [hunk 11]
```diff
@@ -0,0 +1,13 @@
+<?php
+class HomeController extends Controller {
+
+    public function index(){
+
+//        print_r($_SESSION);
```
Comment (dankolosov) [inline Controllers/HomeController.php]: remove commented code

File: Controllers/LoginController.php [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+        Route::redirect('login');
+
+
+    }
+
+
+}
+    $this->View('login');
+    }
+}
```
Comment (dankolosov) [inline Controllers/LoginController.php:48]: remove commented code

File: migrations/001.migration.sql [hunk 13]
```diff
@@ -0,0 +1,4 @@
+SELECT * FROM users;
+dfdfdfdfd
```
Comment (dankolosov) [inline migrations/001.migration.sql]: do you need this?

File: Scripts/connection.php [hunk 14]
```diff
@@ -0,0 +1,14 @@
+<?php
+//class Connection {
```
Comment (dankolosov) [inline Scripts/connection.php]: remove this file

File: Scripts/dataUpdate.php [hunk 15]
```diff
@@ -0,0 +1,18 @@
+<?php
```
Comment (dankolosov) [inline Scripts/dataUpdate.php]: Do you need this file?

File: Scripts/signin.php [hunk 16]
```diff
@@ -0,0 +1,17 @@
+<?php
```
Comment (dankolosov) [inline Scripts/signin.php]: do you need this file?

File: Scripts/signup.php [hunk 17]
```diff
@@ -0,0 +1,35 @@
+<?php
```
Comment (dankolosov) [inline Scripts/signup.php]: Do you need this file?

File: Views/js/skills.js [hunk 18]
```diff
@@ -0,0 +1,145 @@
+
```
Comment (dankolosov) [inline Views/js/skills.js:16]: Wrap code by $(document).ready(function(){})
https://www.w3schools.com/jquery/event_ready.asp

File: Views/js/skills.js [hunk 19]
```diff
@@ -0,0 +1,145 @@
...
+        success: function () {
+            $('#skill' + id).remove();
+        }
+    });
+
+});
+
+
+$(document).on("click", ".close", function(){
```
Comment (dankolosov) [inline Views/js/skills.js]: Remove this

File: Views/js/skills.js [hunk 20]
```diff
@@ -0,0 +1,145 @@
...
+                    $('#skill' + id + ' td:nth-child(2)').text(data[1]);
+                    $('#skill' + id + ' td:nth-child(3)').text(data[2]);
+                    $('#profileEditModal').modal('toggle');
+                    $('#profileEditForm')[0].reset();
+
+                }
+            });
+        }
+
+})
```
Comment (dankolosov) [inline Views/js/skills.js]: Remove all console.log

**Review by dankolosov — APPROVED** [review]

File: Classes/Route.php [hunk 21]
```diff
@@ -0,0 +1,133 @@
+<?php
+session_start();
+
+class Route
+{
+    public static $configurations = [];
+
+    public static $routes = [];
+
+    /**
```
Comment (dankolosov) [inline Classes/Route.php:10]: Here must be a description of the method. But for a private function, you can ignore the commenting method. We comment only public methods.
The same for others

**Review by dankolosov — APPROVED** [review]

**Pull request merged**